### PR TITLE
feat(plasma-website): Redesign for main plasma salute website

### PR DESCRIPTION
**Pull request by neretin-trike**
:baby_chick: Download canary assets:

[colors--canary.437.4543139832.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.437.4543139832.xml)  
[color_metro_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.437.4543139832.swift)  
[color_metro_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.437.4543139832.kt)  
[color_metro_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.437.4543139832.ts)  
[color_sberHealth_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.437.4543139832.swift)  
[color_sberHealth_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.437.4543139832.kt)  
[color_sberHealth_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.437.4543139832.ts)  
[color_sbermarket_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.437.4543139832.swift)  
[color_sbermarket_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.437.4543139832.kt)  
[color_sbermarket_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.437.4543139832.ts)  
[color_sberprime_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.437.4543139832.swift)  
[color_sberprime_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.437.4543139832.kt)  
[color_sberprime_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.437.4543139832.ts)  
[color_selgros_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.437.4543139832.swift)  
[color_selgros_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.437.4543139832.kt)  
[color_selgros_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.437.4543139832.ts)  
[color_smbusiness_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.437.4543139832.swift)  
[color_smbusiness_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.437.4543139832.kt)  
[color_smbusiness_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.437.4543139832.ts)  
[PlasmaTokensColor--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.437.4543139832.swift)  
[shadow_sbermarket_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.437.4543139832.ts)  
[typo_mage_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.437.4543139832.swift)  
[typo_mage_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.437.4543139832.kt)  
[typo_mage_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.437.4543139832.ts)  
[typo_plasma_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.437.4543139832.swift)  
[typo_plasma_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.437.4543139832.kt)  
[typo_plasma_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.437.4543139832.ts)  
[typo_ruler_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.437.4543139832.swift)  
[typo_ruler_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.437.4543139832.kt)  
[typo_ruler_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.437.4543139832.ts)  
[typo_sage_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.437.4543139832.swift)  
[typo_sage_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.437.4543139832.kt)  
[typo_sage_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.437.4543139832.ts)  
[typo_sbermarket_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.437.4543139832.swift)  
[typo_sbermarket_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.437.4543139832.kt)  
[typo_sbermarket_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.437.4543139832.ts)  
[typo_soulmate_ios-swift--canary.437.4543139832.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.437.4543139832.swift)  
[typo_soulmate_kotlin--canary.437.4543139832.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.437.4543139832.kt)  
[typo_soulmate_react-native--canary.437.4543139832.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.437.4543139832.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.167.0-canary.437.4533655578.0
  npm install @salutejs/plasma-hope@0.25.0-canary.437.4533655578.0
  npm install @salutejs/plasma-web@1.192.0-canary.437.4533655578.0
  # or 
  yarn add @salutejs/plasma-b2c@1.167.0-canary.437.4533655578.0
  yarn add @salutejs/plasma-hope@0.25.0-canary.437.4533655578.0
  yarn add @salutejs/plasma-web@1.192.0-canary.437.4533655578.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-437/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-437/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-437/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-437/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-437/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-437/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-437/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: website/plasma-website/components/main/Product.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+`;
+const Card = styled.div`
+    position: relative;
+    width: 100%;
+`;
+const StyledLineSkeleton = styled(LineSkeleton)`
+    width: 6rem;
+`;
+
+const fetchNpmRegistry = async (name: string) => {
```
Comment (Yakutoc) [inline website/plasma-website/components/main/Product.tsx:108]: @neretin-trike Может логику скачивания вынесем отсюда?

File: website/plasma-website/mixins/addGradient.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        background-position: 100%;
+    }
+`;
+
+export const addGradient = css`
+    background: linear-gradient(to right, rgba(255, 255, 255, 0.96) 0% 50%, #2a72f8 100%);
+    background-size: 200%;
+    background-position: 0;
+
+    -webkit-background-clip: text;
```
Comment (Yakutoc) [inline website/plasma-website/mixins/addGradient.ts:17]: @neretin-trike А это будет работать на отличных от **webkit** браузерах?

Comment (neretin-trike) [reply website/plasma-website/mixins/addGradient.ts:17]: я проверял в safari и firefox, всё вроде оке

File: website/plasma-website/components/main/SupportTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    padding: 0.5rem;
+
+    border: 0.063rem solid ${primary};
+    border-radius: 2rem;
+    border-top-right-radius: 0;
+
+    transition: opacity 0.2s ease-in;
+`;
+
+const StyledLink = styled(Link)`
```
Comment (Yeti-or) [inline website/plasma-website/components/main/SupportTooltip.tsx:25]: 😢

File: website/plasma-website/components/main/SupportTooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    `)}
+`;
+
+interface SupportTooltipProps {
+    href?: string;
+}
+
+export const SupportTooltip = ({ href, ...props }: SupportTooltipProps) => (
+    <Root {...props}>
+        <StyledLink as="a" target="_blank" href={href}>
```
Comment (Yeti-or) [inline website/plasma-website/components/main/SupportTooltip.tsx:65]: даже  `as` приходится добавлять?

File: website/plasma-website/pages/index.tsx [hunk 5]
```diff
@@ -39,99 +57,77 @@ const CopyrightText = styled(TextS)`
...
             { text: 'Сторибук', href: `${basePath}/b2c-storybook/` },
-            { text: 'Фигма', href: '', soon: true },
         ],
     },
     {
-        title: 'Web B2B/E',
-        description: 'Для веб-сервисов B2B- и B2E-сегментов',
+        title: 'Web B2E',
+        name: '@salutejs/plasma-web',
+        support: 'https://t.me/smartmarket_community',
```
Comment (Yeti-or) [inline website/plasma-website/pages/index.tsx:85]: нужно поменять на те каналы что у нас есть про support

File: website/plasma-website/pages/index.tsx [hunk 6]
```diff
@@ -39,99 +57,77 @@ const CopyrightText = styled(TextS)`
...
-            </Sidebar>
-            <Main>
-                <Slogan>{slogan}</Slogan>
-                <MainH1 as="h1" my="16x">
-                    {purpose}
-                </MainH1>
-                <MainProducts items={products} />
-            </Main>
-        </StyledContainer>
+                    <CopyrightText>{`©${currentYear} Девайсы`}</CopyrightText>
```
Comment (Yeti-or) [inline website/plasma-website/pages/index.tsx:128]: `СалютДевайсы`

File: website/plasma-website/store/reducer.ts [hunk 7]
```diff
@@ -4,7 +4,7 @@ import { Action, ActionTypes } from './actions';
 import type { State } from './types';
 
 export const initialState: State = {
-    theme: 'light',
+    theme: 'dark',
```
Comment (Yeti-or) [inline website/plasma-website/store/reducer.ts:7]: а оно же у нас зависит от того какая настройка в браузере ?

**Review by Yeti-or — APPROVED** [review]

File: website/plasma-website/components/main/Product.tsx [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    const endpoint = `https://registry.npmjs.org/${name}`;
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply website/plasma-website/components/main/Product.tsx:109]: ага, тоже бы хотелось точечно получать

File: website/plasma-website/store/reducer.ts (hunk 7 above)
Comment (neretin-trike) [reply website/plasma-website/store/reducer.ts:7]: вроде нет, и не уверен, что должна

Comment (Yeti-or) [reply website/plasma-website/store/reducer.ts:7]: @gibushnev у нас есть светлая тема для сайта?

Comment (neretin-trike) [reply website/plasma-website/store/reducer.ts:7]: в макетах отсутствует

Comment (Yeti-or) [reply website/plasma-website/store/reducer.ts:7]: я понимаю, мы в бэклог это положим, я пытаюсь понять нужна ли

Comment (Yeti-or) [reply website/plasma-website/store/reducer.ts:7]: мы же делаем такие классные компоненты и систему что переключение светлой/темной должно быть ну автомагически же

**Comment by kayman233** [comment]
> что-то у меня бетка не открывается нормально Screenshot 2023-03-23 at 13 58 27
> 
> @kayman233 это как-то с nginx связано?

не уверен. Там кажется какие-то проблемы с content-type. Помнится раньше тоже были проблемы такие внутри сторибука, когда при загрузке он автоматически менялся, но мы эту магию с типами убрали тогда

File: website/plasma-website/components/main/SupportTooltip.tsx (hunk 4 above)
Comment (neretin-trike) [reply website/plasma-website/components/main/SupportTooltip.tsx:65]: вообще вроде как не обязательно, вроде ни на что не влияет в данном случае

File: website/plasma-website/pages/index.tsx (hunk 5 above)
Comment (neretin-trike) [reply website/plasma-website/pages/index.tsx:85]: туда нельзя самому добавиться, можешь эту функцию включить и дать ссылочки?

File: website/plasma-website/store/reducer.ts (hunk 7 above)
Comment (neretin-trike) [reply website/plasma-website/store/reducer.ts:7]: в целом идея прикольная, но такого пока у нас нигде не реализовано

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-437/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-437/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-437/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-437/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-437/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-437/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-437/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Yeti-or** [comment]
Screenshot 2023-03-24 at 16 06 03
давайте нахер оторвем эту секцию ?

**Comment by Yeti-or** [comment]
и ешё такой момент, а может мы не будем грузить все метаданные про наши пакеты из npm ;

А прочитаем тупо из репки во время сборки сайта ? типа require('./plasma-ui/package.json')

а за датой можно сходить в релизы github ? https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-a-release-by-tag-name

**Comment by Yeti-or** [comment]
и ещё мы не хотим прописать `font-display: swap` ?? https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display

**Comment by neretin-trike** [comment]
> и ещё мы не хотим прописать `font-display: swap` ?? https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display

но это же нужно делать на уровне `@font-face`, насколько я понял? А мы эти инструкции из cdn тянем

**Comment by neretin-trike** [comment]
> и ешё такой момент, а может мы не будем грузить все метаданные про наши пакеты из npm ;
> 
> А прочитаем тупо из репки во время сборки сайта ? типа require('./plasma-ui/package.json')
> 
> а за датой можно сходить в релизы github ? https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-a-release-by-tag-name

хотел так сначала сделать, но не был уверен, что такой вариант подойдёт. В принципе я могу в сам конфиг next.config.js подргужать скрипт, который выполнится перед сборкой, где вытащу последние версии и запишу их в env, который уже будет в вебе доступен. 

А вместо запроса в гитхаб, могу так же нодой считать дату модификации того же package.json и так же записывать

**Comment by Yeti-or** [comment]
> > и ешё такой момент, а может мы не будем грузить все метаданные про наши пакеты из npm ;
> > А прочитаем тупо из репки во время сборки сайта ? типа require('./plasma-ui/package.json')
> > а за датой можно сходить в релизы github ? https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-a-release-by-tag-name
> 
> хотел так сначала сделать, но не был уверен, что такой вариант подойдёт. В принципе я могу в сам конфиг next.config.js подгружать скрипт, который выполнится перед сборкой, где вытащу последние версии и запишу их в env, который уже будет в вебе доступен.
> 
> А вместо запроса в гитхаб, могу так же нодой считать дату модификации того же package.json и так же сохранять в env

а у нас дока релизится на каждое обновление версии кстати ?

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-437/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-437/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-437/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-437/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-437/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-437/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-437/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-437/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-437/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-437/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-437/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-437/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-437/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by neretin-trike** [comment]
> кажется ПЛАЗМА слишком высоко =(

Высоко потому что грид поломан был, ещё раз надо триггернуть сборку доки

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-437/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-437/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-437/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-437/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-437/temple-storybook/

**Comment by Yeti-or** [comment]
@neretin-trike не помогло

**Comment by neretin-trike** [comment]
> @neretin-trike не помогло

угу, похоже нужно ещё заребезить, чтобы версии поднялись. Сейчас сделаю

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-437/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-437/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-437/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-437/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-437/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-437/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-437/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-437/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-437/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-437/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-437/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-437/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-437/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-437/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Pull request merged**